src/NLog/LayoutRenderers/Log4JXmlEventLayoutRenderer.cs

```diff
@@ -356,7 +357,8 @@ private void AppendNdc(XmlWriter xtw)
                     //extra separator
                     ndcContent += NdcItemSeparator;
                 }
-                ndcContent += string.Join(NdlcItemSeparator, NestedDiagnosticsLogicalContext.GetAllMessages());
+                var ndlcRenderer = new NdlcLayoutRenderer(){Separator = NdlcItemSeparator};
```

Comment:
Instead of allocating a `new NdlcLayoutRenderer` for every LogEvent. Could you put the `_ndlcRenderer` as member-variable, or extract an internal method for rendering the NDLC-string?